lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -740,40 +743,56 @@ monitorMetadata gcStatus tr sp db@(DBLayer{..}) = do
     settings <- atomically readSettings
     manager <- newManager defaultManagerSettings
 
-    let fetcher fetchStrategies = fetchFromRemote trFetch fetchStrategies manager
-        loop getPoolMetadata = forever $ do
-            (refs, successes) <- getPoolMetadata
-            when (null refs || null successes) $ do
-                traceWith tr $ MsgFetchTakeBreak blockFrequency
-                threadDelay blockFrequency
-
-    case poolMetadataSource settings of
-        FetchNone -> do
-            STM.atomically $ writeTVar gcStatus NotApplicable
-            loop (pure ([], [])) -- TODO: exit loop?
-        FetchDirect -> do
-            STM.atomically $ writeTVar gcStatus NotApplicable
-            loop (fetchThem $ fetcher [identityUrlBuilder])
-        FetchSMASH (unSmashServer -> uri) -> do
-            STM.atomically $ writeTVar gcStatus NotStarted
-            let getDelistedPools =
-                    fetchDelistedPools trFetch (toDelistedPoolsURI uri) manager
-            tid <- forkFinally
-                (gcDelistedPools gcStatus tr db getDelistedPools)
-                onExit
-            flip finally (killThread tid) $
-                loop (fetchThem $ fetcher [registryUrlBuilder uri])
+    health <- case poolMetadataSource settings of
+        FetchSMASH uri -> do
+            let checkHealth _ = do
+                    r <- healthCheck (Just trFetch) (unSmashServer uri) manager
```

Comment:
Is it really necessary to run a health check before fetching data?
If the server is healthy it will return data. If it's not healthy it will return some kind of error from the fetch.

Reply:
> Is it really necessary to run a health check before fetching data?

Yes, otherwise you're fetching data from an unknown server that may respond with whatever. SMASH is configurable and we don't control where it points to.

Reply:
Actually the ticket description says "Daedalus has no way to know whether such a [SMASH] url points to a valid server."
So this change should just provide a way for Daedalus to check that a SMASH URL is correct before applying the setting.

There is no real need to call the health check API before fetching data. Why does everything become more complicated when a health check function is added?

Reply:
> So this change should just provide a way for Daedalus to check that a SMASH URL is correct before applying the setting.

The ticket says that we want to know whether it points to a valid server, not whether the URL is valid. That's not the same thing. We are dealing with user input here. We don't want to start fetching stuff randomly and then show errors. We want to know ahead of time whether a URL points to an actual SMASH server and that server is ready to accept requests. And if it isn't, daedalus can reject the setting. If we just apply it and then error out when actual metadata is fetched, then that's poor user experience.